tests/jballerina-unit-test/src/test/resources/test-src/annotations/annotation_readonly_types.bal

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.'object as lang;
+import ballerina/java;
+
+type Annot record {|
+   service myService;
+   object {} myObject;
+   handle myHandle;
+   typedesc myTypedesc;
+   function () myFunction;
+|};
+
+public annotation Annot v1 on type;
+
+string strValue = "v1 value";
+
+type Dummy abstract object {};
+
+type DummyImpl object {
+    *Dummy;
+};
+
+public function newArrayList() returns handle = @java:Constructor {
+    class:"java.util.ArrayList"
+} external;
+
+handle dummyHandle = newArrayList();
+
+typedesc<any> T = typeof strValue;
+
+var fn = function () {
+    int x = 5;
+    int y = 4;
+};
+
+DummyImpl dummyImpl = new();
+
+@v1 {
+    myService: ser,
+    myObject: dummyImpl,
+    myHandle: dummyHandle,
+    myTypedesc: T,
+    myFunction: fn
+}
+public type T1 record {
+    string name;
+};
+
+T1 a = { name: "John" };
+
+function testReadonlyTypeAnnotationAttachment()  {
+    typedesc<any> t = typeof a;
+    Annot? annot = t.@v1;
+    assertTrue(annot is Annot);
+}
+
+public function main() {
+    testTypeAnnotAccess1();
```

Comment:
Does this compile? Can't seem to find `testTypeAnnotAccess1`.

We don't need this main function anyway right?

Reply:
No..I have run locally disabling the test suite.
Fixed: 5604d29b2fc131265541d30ea1335c316fb8fe82